### PR TITLE
Update Sulu 2.5 and Support Symfony 6

### PR DESCRIPTION
**Pull request by TheCadien**

### CODE REVIEW

**Comment by TheCadien** [comment]
Need to fix at next 
`The controller for URI "/_sulu_preview" is not callable: Controller "sulu_news.controller:indexAction" does neither exist as service nor as class.`

**Comment by alexander-schranz** [comment]
@TheCadien should mostly be change route definition to `::`:  `sulu_news.controller:indexAction` -> `sulu_news.controller::indexAction` ;)

**Comment by TheCadien** [comment]
@alexander-schranz your absolute right ! thanks !! :)

**Comment by alexander-schranz** [comment]
@TheCadien your dev dependencies blocks the updates in the CI.

You need to update phpstan and php-cs-fixer so CI runs on Symfony 6. Currently it runs in mixed mode which make things failing:

Bildschirmfoto 2022-11-07 um 16 50 06

File: Twig/NewsTwigExtension.php [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+ *
+ * (c) Oliver Kossin
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace TheCadien\Bundle\SuluNewsBundle\Twig;
+
+use Doctrine\Common\Cache\Cache;
```
Comment (dbrumann) [inline Twig/NewsTwigExtension.php:16]: Doctrine Cache is deprecated. See: https://www.doctrine-project.org/projects/doctrine-cache/en/2.2/index.html

You should consider moving to either a PSR-6 Cache or using Symfony's Cache contract.

File: Twig/NewsTwigExtension.php [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    /**
+     * @var Cache
+     */
+    private $cache;
+
+    public function __construct(Cache $cache, NewsRepository $newsRepository)
+    {
+        $this->cache = $cache;
+        $this->newsRepository = $newsRepository;
+    }
```
Comment (dbrumann) [inline Twig/NewsTwigExtension.php:40]: According to composer.json PHP 8.0 is the minimum required version. You could simplify this to:

```suggestion
    public function __construct(
        private readonly Cache $cache,
        private readonly NewsRepository $newsRepository
    ) {}
```

File: Twig/NewsTwigExtension.php [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    {
+        return [
+            new TwigFunction('sulu_resolve_news', [$this, 'resolveNewsFunction']),
+        ];
+    }
+
+    /**
+     * @param int $id id to resolve
+     */
+    public function resolveNewsFunction($id)
```
Comment (dbrumann) [inline Twig/NewsTwigExtension.php:52]: Again, modern PHP allows you to replace doc block with proper type declaration:

```suggestion
    public function resolveNewsFunction(int $id): ?string
```

File: Twig/NewsTwigExtension.php [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    {
+        if ($this->cache->contains($id)) {
+            return $this->cache->fetch($id);
+        }
+
+        $news = $this->newsRepository->find($id);
+        if (null === $news) {
+            return;
```
Comment (dbrumann) [inline Twig/NewsTwigExtension.php:60]:
```suggestion
            return null;
```

**Comment by Basster** [comment]
Hi.

I'm just testing the bundle with this branch and I came up with an issue:

I've created a pretty simple `news/index.html.twig`:

```twig
{% block content %}
    <h2>{{ news.title }}</h2>

    {% set header = sulu_resolve_media(news.header, 'de') %}
    <img src="{{ header.thumbnails['sulu-260x'] }}" alt="{{ header.title }}" title="{{ header.title }}" />

    <p>{{ news.teaser }}</p>
    
    {% for contentItem in news.content  %}
        {% if contentItem.type == 'editor'  %}
            <p>{{ contentItem.text | raw }}</p>
        {% endif %}
    {% endfor %}
{% endblock %}
```
 In the preview pane I get the following error:

```
Sulu\Bundle\PreviewBundle\Preview\Exception\TwigException:
An exception has been thrown during the rendering of a template ("Warning: foreach() argument must be of type array|object, null given"). (/Users/oroessner/Development/my/project/templates/news/index.html.twig line 5)

  at vendor/sulu/sulu/src/Sulu/Bundle/PreviewBundle/Preview/Renderer/PreviewRenderer.php:199
  at Sulu\Bundle\PreviewBundle\Preview\Renderer\PreviewRenderer->render(object(News), '1', false, array('webspaceKey' => 'my-webspace', 'locale' => 'de', 'targetGroupId' => '-1'))
     (vendor/sulu/sulu/src/Sulu/Bundle/PreviewBundle/Preview/Preview.php:152)
  at Sulu\Bundle\PreviewBundle\Preview\Preview->render('3a5782230389e5b89c95f4e109b59942', array('webspaceKey' => 'my-webspace', 'locale' => 'de', 'targetGroupId' => '-1'))
     (vendor/sulu/sulu/src/Sulu/Bundle/PreviewBundle/UserInterface/Controller/PreviewController.php:78)
  at Sulu\Bundle\PreviewBundle\UserInterface\Controller\PreviewController->renderAction(object(Request))
     (vendor/symfony/http-kernel/HttpKernel.php:163)
  at Symfony\Component\HttpKernel\HttpKernel->handleRaw(object(Request), 1)
     (vendor/symfony/http-kernel/HttpKernel.php:75)
  at Symfony\Component\HttpKernel\HttpKernel->handle(object(Request), 1, true)
     (vendor/symfony/http-kernel/Kernel.php:202)
  at Symfony\Component\HttpKernel\Kernel->handle(object(Request))
     (public/index.php:68) 
```

If I dig deeper, its in [\Sulu\Bundle\MediaBundle\Api\Media::getFileVersion()](https://github.com/sulu/sulu/blob/2.5/src/Sulu/Bundle/MediaBundle/Api/Media.php#L925) `$this->entity->getFiles()` is empty. Don't know if its related to the media bundle or your `\TheCadien\Bundle\SuluNewsBundle\Controller\NewsWebsiteController::renderPreview`.

But if I publish it or create a preview and open it in the browser, it renders fine. 

grafik

package-versions:
* sulu/sulu: 2.5.3
* thecadien/sulu-news-bundle: dev-sulu25 (15555e14ecc17be1e868aa1cce3dd1ae67b7760f)

**Comment by TheCadien** [comment]
@Basster , thanks for reporting. 
Please try it again with the ID not the Media Object. Like that way

`{% block content %}
    <h2>{{ news.title }}</h2>

    {% set header = sulu_resolve_media(news.header.id, 'de') %}
    {{ header.title }}

    <p>{{ news.teaser }}</p>
    
    {% for contentItem in news.content  %}
        {% if contentItem.type == 'editor'  %}
            <p>{{ contentItem.text | raw }}</p>
        {% endif %}
    {% endfor %}
{% endblock %}`

**Comment by Basster** [comment]
> Please try it again with the ID not the Media Object. Like that way

@TheCadien works like a charm. Thank you.

**Pull request merged**